src/components/legend.ts

```diff
@@ -144,13 +164,17 @@ export module Components {
 
       let availableWidthForEntries = Math.max(0, (availableWidth - this._padding));
 
-      let entryNames = this._colorScale.domain().slice();
-      entryNames.sort(this.comparator());
+      let entryLabels: d3.Map<string> = d3.map<string>();
+      this._colorScale.domain().slice().forEach((d) => {
+        entryLabels.set(this._formatter(d), d);
+      });
+      let entryNames = entryLabels.keys();
+      entryNames = entryNames.sort(this.comparator()).map((d) => entryLabels.get(d));
```

Comment:
Please check if this is what people want first. If this is indeed the case, you can map the formatter over all of the labels and then apply the sort then

`entryNames.map(this.formatter()).sort(this.comparator())`